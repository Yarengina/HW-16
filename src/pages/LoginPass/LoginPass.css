.Form {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-top: 30px;
}

.Inputs {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
}

.Input {
    width: 200px;
    height: 35px;
    border: 1px solid rgb(153, 153, 153);
    border-radius: 4px;
    font-size: 15px;
    padding: 0px 15px;
    margin-top: 20px;
}

.Button {
    width: 230px;
    height: 38px;
    background-color: rgb(60, 65, 180);
    color: white;
    border: 1px solid rgb(153, 153, 153);
    border-radius: 4px;
    font-size: 18px;
    padding: 0px 15px;
    margin-top: 30px;
    margin-bottom: 20px;
}

.Error {
    color: red;
    font-size: 14px;
}
