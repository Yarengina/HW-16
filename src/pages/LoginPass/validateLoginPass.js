import { validEmail, passwordLength } from '../../components/constants'

export default function validateLoginPass(userEmail, userPassword) {
    if (userEmail.match(validEmail) && userPassword.length >= passwordLength) {
        return true
    }
    return false
}
