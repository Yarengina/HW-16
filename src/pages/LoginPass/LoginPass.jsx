// eslint-disable-next-line no-unused-vars
import * as React from 'react'
import { useRef, useContext } from 'react'
import { useNavigate } from 'react-router-dom'
import useCheckInput from '../../components/useCheckInput'
import UserInfoContext from '../../contexts/UserInfoContext'
import './LoginPass.css'
import Navigation from '../../components/Navigation/Navigation'
import validateLoginPass from './validateLoginPass'

function LoginPass() {
    const { inputValue, onChange, onBlur, isEmailError, isPassError } =
        useCheckInput('')

    const { user, setUser } = useContext(UserInfoContext)

    const navigate = useNavigate()

    const inputEmail = useRef(null)
    const inputPassword = useRef(null)
    const emailError = useRef(null)
    const passwordError = useRef(null)

    const onSubmit = (event) => {
        event.preventDefault()
    }

    const handleClick = () => {
        const userEmail = inputEmail.current.value
        const userPassword = inputPassword.current.value

        if (inputEmail.current.name === 'email' && userEmail === '') {
            inputEmail.current.focus()
            emailError.current.innerText = 'Type your e-mail'
            return
        }

        if (inputPassword.current.name === 'password' && userPassword === '') {
            inputPassword.current.focus()
            passwordError.current.innerText = 'Type your password'
            return
        }

        if (validateLoginPass(userEmail, userPassword)) {
            navigate('/order')
            setUser({
                ...user,
                email: userEmail,
            })
        }
    }

    return (
        <div>
            <Navigation />
            <form className="Form" onSubmit={onSubmit}>
                <div className="Inputs">
                    <input
                        className="Input"
                        id="1"
                        name="email"
                        type="email"
                        placeholder="Enter your email..."
                        value={inputValue.inputEmail}
                        onChange={onChange}
                        onBlur={onBlur}
                        ref={inputEmail}
                    />
                    <span className="Error" ref={emailError}>
                        {isEmailError}
                    </span>
                </div>
                <div className="Inputs">
                    <input
                        className="Input"
                        id="2"
                        name="password"
                        type="password"
                        placeholder="Enter your password..."
                        value={inputValue.inputPassword}
                        onChange={onChange}
                        onBlur={onBlur}
                        ref={inputPassword}
                    />
                    <span className="Error" ref={passwordError}>
                        {isPassError}
                    </span>
                </div>
                <button className="Button" type="button" onClick={handleClick}>
                    Log in
                </button>
            </form>
        </div>
    )
}

export default LoginPass
