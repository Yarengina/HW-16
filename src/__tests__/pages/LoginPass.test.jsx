// eslint-disable-next-line no-unused-vars
import * as React from 'react'
import '@testing-library/jest-dom/extend-expect'
import { screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { render } from '../utils/renderWithRouter'
import LoginPass from '../../pages/LoginPass/LoginPass'

describe('<LoginPass />', () => {
    it('should render successfully and match snapshot', () => {
        const { container } = render(<LoginPass />)
        expect(container).toMatchSnapshot()
    })

    it('should test initially empty inputs', () => {
        render(<LoginPass />)
        const inputEmail = screen.getByPlaceholderText(/enter your email.../i)
        const inputPassword = screen.getByPlaceholderText(
            /enter your password.../i
        )

        expect(inputEmail.value).toBe('')
        expect(inputPassword.value).toBe('')
    })

    it('should be able to type an email', () => {
        render(<LoginPass />)
        const inputEmail = screen.getByPlaceholderText(/enter your email.../i)
        userEvent.type(inputEmail, 'user@gmail.com')
        expect(inputEmail.value).toBe('user@gmail.com')
    })

    it('should be able to type a password', () => {
        render(<LoginPass />)
        const inputPassword = screen.getByPlaceholderText(
            /enter your password.../i
        )
        userEvent.type(inputPassword, 'Password!')
        expect(inputPassword.value).toBe('Password!')
    })

    it('should show email error message on invalid email', () => {
        render(<LoginPass />)
        const inputEmail = screen.getByPlaceholderText(/enter your email.../i)
        userEvent.type(inputEmail, 'usergmail.com')

        const logInButton = screen.getByRole('button', { name: /log in/i })
        userEvent.click(logInButton)

        const emailError = screen.getByText(/email is not valid/i)
        expect(emailError).toBeInTheDocument()
    })

    it('should show password error message on invalid password', () => {
        render(<LoginPass />)
        const inputPassword = screen.getByPlaceholderText(
            /enter your password.../i
        )
        userEvent.type(inputPassword, '1234')

        const logInButton = screen.getByRole('button', { name: /log in/i })
        userEvent.click(logInButton)

        const passwordError = screen.getByText(/at least 6 characters/i)
        expect(passwordError).toBeInTheDocument()
    })
})
