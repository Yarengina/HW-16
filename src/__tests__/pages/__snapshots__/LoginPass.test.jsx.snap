// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPass /> should render successfully and match snapshot 1`] = `
<div>
  <div>
    <nav
      class="LinkField"
    >
      <a
        aria-current="page"
        class="Link active"
        data-testid="login-link"
        href="/"
      >
        <span>
          Login
        </span>
      </a>
      <a
        class="Link"
        data-testid="about-link"
        href="/about"
      >
        <span>
          About
        </span>
      </a>
      <a
        class="Link"
        data-testid="order-link"
        href="/order"
      >
        <span>
          Order
        </span>
      </a>
    </nav>
    <form
      class="Form"
    >
      <div
        class="Inputs"
      >
        <input
          class="Input"
          id="1"
          name="email"
          placeholder="Enter your email..."
          type="email"
          value=""
        />
        <span
          class="Error"
        />
      </div>
      <div
        class="Inputs"
      >
        <input
          class="Input"
          id="2"
          name="password"
          placeholder="Enter your password..."
          type="password"
          value=""
        />
        <span
          class="Error"
        />
      </div>
      <button
        class="Button"
        type="button"
      >
        Log in
      </button>
    </form>
  </div>
</div>
`;
